src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    this.name = name;
+    this.health = health;
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health, hidden);
+    this.hidden = false;
```

Comment:
It's unnecessary to set 'this.hidden = false;' in the Herbivore constructor since it's already being set in the Animal constructor. However, since 'hidden' should not be in the Animal constructor, this line should be kept and the 'hidden' parameter removed from the Animal constructor.